File: pinyougou-user-interface/src/main/java/com/pinyougou/user/service/UserMessageService.java
package com.pinyougou.user.service;

import com.pinyougou.pojo.TbAreas;
import com.pinyougou.pojo.TbCities;
import com.pinyougou.pojo.TbProvinces;
import com.pinyougou.pojo.group.UserMessage;
import entity.Result;

import java.util.List;

public interface UserMessageService {
    /**
     *
     * 添加个人信息
     * @return
     */
    public void add(UserMessage userMessage);

    /**
     * 查询省份地址列表
     * @return
     */
    public List<TbProvinces> findAddressList();

    /**
     * 根据省份id查询市级列表
     * @return
     */
    public List<TbCities> findCities(String id);

    /**
     * 根据市级id查询地区列表
     * @param cityId
     * @return
     */
    public List<TbAreas> findAreas(String cityId);

    /**
     * 根据用户登录名查询用户
     * @param username
     * @return
     */
    public UserMessage findOneUser(String username);
}

File: pinyougou-user-web/src/main/webapp/js/service/userMessageService.js
//服务层
app.service('userMessageService',function($http){

	//添加个人信息
	this.add=function (userMessage) {
		return $http.post('../message/add.do',userMessage);
    }

    //查找地址集合
	this.findAddressList=function () {
		return $http.post('../message/findAddressList.do')
    }

    //查询市级列表
    this.findCities=function (provinceId) {
		return $http.post('../message/findCities.do?provinceId='+provinceId)
    }


    this.findAreas=function (cityId) {
		return $http.post('../message/findAreas.do?cityId='+cityId)
    }

    this.findOneUser=function (loginName) {
        return $http.post('../message/findOneUser.do?loginName='+loginName)
    }
});
